src/cascadia/ElevateShim/elevate-shim.cpp

```diff
@@ -46,8 +99,10 @@ int __stdcall wWinMain(HINSTANCE, HINSTANCE, LPWSTR pCmdLine, int)
     seInfo.cbSize = sizeof(seInfo);
     seInfo.fMask = SEE_MASK_DEFAULT;
     seInfo.lpVerb = L"runas"; // This asks the shell to elevate the process
-    seInfo.lpFile = module.c_str(); // This is `...\WindowsTerminal.exe`
-    seInfo.lpParameters = pCmdLine; // This is `new-tab -p {guid}`
+    seInfo.lpFile = cmd.c_str(); // This is `shell:AppsFolder\...` or `...\WindowsTerminal.exe`
+    seInfo.lpParameters = GetCommandLine(); // This is `new-tab -p {guid}`

```

Comment:
Why not use the previous `pCmdLine`?

Reply:
I found an innocuous bug in how the first parameter, `new-tab`, was dropped when passed to elevate-shim.exe by wWinMain given how it's launched by `CreateProcess`. It appears wWinMain is assuming param zero is the executable path even if it's not.

However, the elevate-shim process is launched with a command line that doesn't include the executable path.

https://github.com/microsoft/terminal/blob/8e041692b3321e2dd31106f4d9096673eb43b0e7/src/cascadia/TerminalApp/TerminalPage.cpp#L4126-L4135

---

This is the !peb from elevate-shim.exe when launched from a dev build of terminal as an appx (I setup Windbg Preview as the post-mortem debugger and put a DebugBreak() at the top of elevate-shim to break in easily). Notice the `CommandLine` is missing the executable filepath as the first argument. You can also see this in Process Monitor.

```
    CurrentDirectory:  'C:\windows\system32\'
    WindowTitle:  'C:\source\github.com.jboelter.terminal\src\cascadia\CascadiaPackage\bin\x64\Debug\AppX\elevate-shim.exe'
    ImageFile:    'C:\source\github.com.jboelter.terminal\src\cascadia\CascadiaPackage\bin\x64\Debug\AppX\elevate-shim.exe'
    CommandLine:  '  new-tab --profile "{0caa0dad-35be-5f56-a8ff-afceeeaa6101}"'
    DllPath:      'C:\source\github.com.jboelter.terminal\src\cascadia\CascadiaPackage\bin\x64\Debug\AppX;C:\Program Files\WindowsApps\Microsoft.VCLibs.140.00.Debug.UWPDesktop_14.0.30704.0_x64__8wekyb3d8bbwe;'
 ```

Reply:
Yeeeeeeeep that's totally on me. We probably could fix that down in TerminalPage.cpp, but 🤷